src/utils/tracked-controls.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * Return enumerated gamepads matching id prefix.
+ *
+ * @param {object} idPrefix - prefix to match in gamepad id, if any.
+ */
+module.exports.getGamepadsByPrefix = function (idPrefix) {
+  var gamepadsList = [];
+  var gamepad;
+  var gamepads = navigator.getGamepads && navigator.getGamepads();
+  if (!gamepads) { return gamepadsList; }
+
+  for (var i = 0; i < gamepads.length; ++i) {
+    gamepad = gamepads[i];
+    if (gamepad) {
```

Comment:
initially navigator.getGamepads() returns an array of 4 null values which is why this is necessary